### DISCUSSION TITLE
Getting next-auth userId inside getStaticProps

### DISCUSSION THREAD

**Discussion by elfraiman** [prisma/prisma#6445] [Q&A] [answered]
Hey 👋 !

I'm trying to figure out how I can get the current sessions userId inside the getStaticProps so I can use that id to fetch a dataset from prisma.

Any quick solutions here?

Here's an example

```ts
export const getStaticProps: GetStaticProps = async () => {
  const userId = ???? // cant useSession inside staticprops

  const player = await prisma.player.findUnique({
    where: { userId: userId},
  });

  return { props: { player } };
};
```

**Comment by ryands17** [comment, marked as answer]
@elfraiman 👋 

This [link](https://next-auth.js.org/tutorials/securing-pages-and-api-routes#server-side) should help 👍

**Reply by elfraiman** [reply]
Thanks a bunch!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
